### PR TITLE
add module picard/collectrnaseqmetrics

### LINKED ISSUE #2652: new module: gatk4/collectrnaseqmetrics
### Is there an existing module for this?

- [X] I have searched for the existing module

### Is there an open PR for this?

- [X] I have searched for existing PRs

### Is there an open issue for this?

- [X] I have searched for existing issues

### Are you going to work on this?

- [X] If I'm planning to work on this module, I added myself to the `Assignees` to facilitate tracking who is working on the module

### PR DESCRIPTION
**Pull request by anoronh4**
Adding the module picard/collectrnaseqmetrics. allowed usage of a gene_pred file as an alternative to ref_flat. On further consideration, there are a few other strategies i can think of to generate the ref_flat file, but maybe the nf-core community should weigh in here.  

## PR checklist

Closes #2652 

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfill software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

**Comment by anoronh4** [comment]
a few strategies to provide necessary ref_flat file for collectrnaseqmetrics:
1. add file to nf-core/test-datasets repo
2. create a new module specifically to this purpose called ucsc/gtftorefflat. This would be a lot like the existing ucsc/gtftogenepred but would add an extra awk statement at the end to create a refflat file
3. have the collectrnaseqmetrics module take a gene_pred file as an alternative to the ref_flat file
4. modify the existing ucsc/gtftogenepred module to output an additional refflat channel.

**Comment by anoronh4** [comment]
> At the time of writing this comment I am stuck on the best way to add provide a ref_flat file for this module. a few strategies i've thought of include:
> 
> 1. add file to the `nf-core/test-datasets` repo
> 2. create a new module specifically to this purpose called `ucsc/gtftorefflat`. This would be a lot like the existing `ucsc/gtftogenepred` but would add an extra awk statement at the end to create a ref_flat file
> 3. have the `picard/collectrnaseqmetrics` module take a gene_pred file as an alternative to the ref_flat file
> 4. modify the existing `ucsc/gtftogenepred` module to output an additional channel containing the ref_flat file.

I have chosen the fourth option and the PR for updating `ucsc/gtftogenepred` was already created and merged (#2680)

**Review by nvnieuwk — CHANGES_REQUESTED** [review]

File: modules/nf-core/picard/collectrnaseqmetrics/main.nf [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    def rrna = rrna_intervals ? "--RIBOSOMAL_INTERVALS ${rrna_intervals}" : ""
+    def strand = ""
+    if ( meta.strandedness == "forward" || meta.single_end ) {
+        strand = "--STRAND_SPECIFICITY FIRST_READ_TRANSCRIPTION_STRAND"
+    } else if (meta.strandedness == "forward") {
+        strand = "--STRAND_SPECIFICITY SECOND_READ_TRANSCRIPTION_STRAND"
+    } else { strand = "--STRAND_SPECIFICITY NONE" }
+    def avail_mem = 3
+    if (!task.memory) {
+        log.info '[Picard CollectMultipleMetrics] Available memory not known - defaulting to 3GB. Specify process memory requirements to change this.'
```
[the remark this answers is not in the document]
Comment (anoronh4) [reply modules/nf-core/picard/collectrnaseqmetrics/main.nf]: Fixed, thanks for catching that!

File: modules/nf-core/picard/collectrnaseqmetrics/main.nf [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    def reference = fasta ? "--REFERENCE_SEQUENCE ${fasta}" : ""
+    def rrna = rrna_intervals ? "--RIBOSOMAL_INTERVALS ${rrna_intervals}" : ""
+    def strand = ""
+    if ( meta.strandedness == "forward" || meta.single_end ) {
+        strand = "--STRAND_SPECIFICITY FIRST_READ_TRANSCRIPTION_STRAND"
+    } else if (meta.strandedness == "forward") {
+        strand = "--STRAND_SPECIFICITY SECOND_READ_TRANSCRIPTION_STRAND"
+    } else { strand = "--STRAND_SPECIFICITY NONE" }
```
[the remark this answers is not in the document]
Comment (anoronh4) [reply modules/nf-core/picard/collectrnaseqmetrics/main.nf]: i removed these statements and actually added `STRAND_SPECIFICITY` in the test config file.

**Review by nvnieuwk — APPROVED** [review]
LGTM!

**Pull request merged**